### DISCUSSION TITLE
Make blocks availaible for Winter.Blog plugin

### DISCUSSION THREAD

**Discussion by damsfx** [wintercms/wn-blocks-plugin#14] [Q&A]
I've tried replacing the blog editor with blocks.  
This try is currently available as a plugin, not sure if it would be of interest to users of the Winter.Blocks plugin.

The plugin is available here : [Blocks for Blog plugin](https://github.com/Hounddd/wn-blocksforblog-plugin)

At this time : 
- Replace post editor by blocks if enable via extended Blog settings.
- Add a special block to use already uploaded **featured_images** attachments in content with a dedicated dropdown 
![image](https://github.com/wintercms/wn-blocks-plugin/assets/282242/025427da-bf52-4f91-a935-d12fa4efe849)
- Save the blocks definition in post metadata and convert blocks render to HTML and markdown for backward compatibility if this plugin is disable (or post editor replacement setting is disable)

That's say, I was wondering if it was possible to create a block that could use **content_images** attachments by offering a drop zone for downloading, replaced by the image afterwards (as in the split base editor's preview mode) ?

**Comment by helmutkaufmann** [comment]
I had done one more with several features, such as Ajax-based reloading of the next articles and "read more". The code is here: https://github.com/[helmutkaufmann/wn-blocksblog-plugin](https://github.com/helmutkaufmann/wn-blocksblog-plugin) but be aware that the .htm files are using getuikit.com

Example can be found on teh front page of [obergrund.ch](obergrund.ch)

**Reply by damsfx** [reply]
@helmutkaufmann Can you explain why you redefined all the blog models and controllers rather than creating an extension for the existing plugin?  
I find it a little ambiguous for the user.

Another question, where is defined the [`dynComponentRender`](https://github.com/helmutkaufmann/wn-blocksblog-plugin/blob/3de087d4553ae527b7ba4d3cf0957994c34ae032/blocks/blog_posts_list.block#L104) twig function?

**Reply by helmutkaufmann** [reply]
I was looking for a block version of the blog plugin and did not see that there was an extension already. It was a weeked morning and I was playing around with an AI model (which failed the original task of answering regulatory questions). So I asked it to do a blog plugin to test coding abilities and as I liked the initial result l kept it. Also, i wanted blocks for both, the teaser and the content, not just for the content itself.
I find it a little ambiguous for the user: right. and I am probaly the only users besides the users for the website
for your other question: [https://github.com/helmutkaufmann/wn-dynblocks-plugin](https://github.com/helmutkaufmann/wn-dynblocks-plugin)

**Reply by damsfx** [reply]
> ...  your other question: https://github.com/helmutkaufmann/wn-dynblocks-plugin

OK. So perhaps it should then be included in your [composer requirements](https://github.com/helmutkaufmann/wn-blocksblog-plugin/blob/3de087d4553ae527b7ba4d3cf0957994c34ae032/composer.json#L19-L22).

**Reply by helmutkaufmann** [reply]
Thanks, @damsfx, will do.
